### PR TITLE
feat: support feature flags

### PR DESCRIPTION
**Pull request by vtsaplin**

### CODE REVIEW

File: web-src/src/components/ApplicationProvider.js [hunk 1]
```diff
@@ -36,54 +37,67 @@ export const ApplicationContext = React.createContext(undefined);
...
+    const createApplicationContext = async () => {
+      console.debug('Creating application context...');
+
+      const featureFlags = await getFeatureFlags();
+      console.debug('Feature flags:', featureFlags);
+
+      return {
+        appVersion: APP_VERSION,
+
+        featureFlags,
```
Comment (basecode) [inline web-src/src/components/ApplicationProvider.js:61]: for consistency, what about..?

```
featureFlagService: new FeatureFlagService({
   projectId: 'aem-generate-variations'
}),
```

where `featureFlagService` exposes  a get method: `featureFlagService.get('SITES-1234')` which returns a boolean.

File: web-src/src/helpers/FeatureFlagsHelper.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import featureFlags from '@adobe/exc-app/featureflags';
+
+const PROJECT_ID = 'aem-generate-variations';
+
+export async function getFeatureFlags() {
+  const flags = await featureFlags.get([PROJECT_ID]);
+  return flags[PROJECT_ID];
```
Comment (basecode) [inline web-src/src/helpers/FeatureFlagsHelper.js:18]: Does the contract of `@adobe/exc-app/featureflags` really asks you twice for the project id? Appears redudant. I see it being used in L17 and L18.

Comment (askayastha22) [reply web-src/src/helpers/FeatureFlagsHelper.js:18]: It's strange but it works. Just tried it myself.

File: web-src/src/components/ApplicationProvider.js (hunk 1 above)
Comment (askayastha22) [reply web-src/src/components/ApplicationProvider.js:61]: Agree with @basecode

File: web-src/src/services/FeatureFlagsService.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+export class FeatureFlagsService {
+  constructor(flags = {}) {
+    this.flags = flags;
+  }
+
+  isEnabled(flagName) {
+    return this.flags[flagName];
+  }
+
+  static async create(projectId) {
```
[the remark this answers is not in the document]
Comment (vtsaplin) [reply web-src/src/services/FeatureFlagsService.js:23]: This pattern involves a static factory method that functions like a constructor. The creation of the instance is async. In Java, we would typically make the constructor private in such cases.

File: web-src/src/services/FeatureFlagsService.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    const flags = await featureFlags.get([projectId]);
+    const projectFlags = flags[projectId];
+    console.debug('Feature flags:', projectFlags);
```
[the remark this answers is not in the document]
Comment (vtsaplin) [reply web-src/src/services/FeatureFlagsService.js:34]: I would say yes, it's intentional for simpler troubleshooting.

File: web-src/src/services/FeatureFlagsService.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (vtsaplin) [reply web-src/src/services/FeatureFlagsService.js:31]: Added a comment.

**Pull request merged**